cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -884,6 +891,56 @@ public void onDeny() {
         });
   }
 
+  /**
+   * For geofencing, ACCESS_BACKGROUND_LOCATION permission is needed starting from Android 10 (API level 29).
+   * This is because geofencing is a feature that typically requires app to receive location updates
+   * even when it is not in the foreground, which qualifies as background location access.
+   */
+  private void showSettingsDialog(Activity activity) {
+    try {
+      Locale currentLocale = Locale.getDefault();
+      String title, message, positiveButton, negativeButton;
+
+      if ("de".equals(currentLocale.getLanguage())) {
+        title = "Erlaubnis erforderlich";
+        message = "Diese App benötigt Hintergrundzugriff auf den Standort, um Geofencing-Funktionen zu aktivieren. Bitte aktivieren Sie diese Berechtigung in den App-Einstellungen." +
```

Comment:
Put translations into constants so we can easily extend to more translations in the future.
And please also put the existing translations we have from the SDK into this constant / class for getting the translated value.
So we can call something like `CPTranslations.get('locationPermissionDialog.title')` (pseudo code)